syft/frameworks/torch/tensors/interpreters/native.py

```diff
@@ -742,27 +765,37 @@ def share(
                 syft.AdditiveSharingTensor(
                     field=field, crypto_provider=crypto_provider, owner=self.owner
                 )
-                .on(self.copy())
-                .child.init_shares(*owners)
+                .on(self.copy(), wrap=False)
```

Comment:
While we're here, have you checked to see if this pattern is anywhere else in the code? it seems like a good time to fix them all

Reply:
Looks good!